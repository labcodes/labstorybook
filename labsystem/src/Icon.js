import React from "react";
import PropTypes from "prop-types";

export default class Icon extends React.Component {
  static propTypes = {
    type: PropTypes.oneOf([
      "arrow-down",
      "arrow-left",
      "arrow-right",
      "arrow-top",
      "calendar",
      "coin",
      "collapse-closed",
      "collapse-open",
      "check",
      "dropdown-closed",
      "dropdown-open",
      "edit",
      "eye-closed",
      "eye-opened",
      "track",
      "key",
      "logout",
      "lupe",
      "minus",
      "plus",
      "reload",
      "remove",
      "sort",
      "trash",
      "upload",
    ]).isRequired,
    color: PropTypes.oneOf([
      "white",
      "teal40",
      "mineral70",
      "mineral20"
    ]).isRequired,
    size: PropTypes.oneOf(["small", "petit"]),
  };

  render() {
    const { type, color, size } = this.props;
    return <span className={`lab-icon ${type} lab-icon--${color} lab-icon--${size}`} />;
  }
}
