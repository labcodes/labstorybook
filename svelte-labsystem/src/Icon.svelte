<script>
  export let type = undefined;
  export let size = undefined;
  export let color = undefined;

  let className = '';
  export { className as class };
</script>

<span
  class={`lab-icon lab-icon--${type} lab-icon--${color}` + `${size ? ` lab-icon--${size}` : ''}` + ` ${className}`} />
