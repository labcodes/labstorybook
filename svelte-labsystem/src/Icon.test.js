import { render } from "@testing-library/svelte";
import Icon from "./Icon.svelte";

describe("Icon", () => {
  it("renders with base props", async () => {
    const { container } = render(Icon);
    expect(container).toMatchSnapshot();
  });

  it("renders with specific props", async () => {
    const { container } = render(Icon, {
      props: {
        class: "test",
        color: "teal-30",
        type: "edit",
        size: "large",
      },
    });

    expect(container).toMatchSnapshot();
  });
});
