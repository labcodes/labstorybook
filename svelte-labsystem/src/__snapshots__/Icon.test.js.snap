// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders with base props 1`] = `
<body>
  <div>
    <span
      class="lab-icon lab-icon--undefined lab-icon--undefined "
    />
  </div>
</body>
`;

exports[`Icon renders with specific props 1`] = `
<body>
  <div>
    <span
      class="lab-icon lab-icon--edit lab-icon--teal-30 lab-icon--large test"
    />
  </div>
</body>
`;
